### PR TITLE
Allow speakers to update their approved talks. (fix #482)

### LINKED ISSUE #482: Some abstracts are not editable
@EuroPython/webwg 
We received some reports that the speakers are not able to edit the submissions.
see e.g. helpdesk:
https://helpdesk.europython.eu/tickets/20422/

### PR DESCRIPTION
**Pull request by umgelurgel**
- [x]  add tests

fixes #482

### CODE REVIEW

File: conference/talks.py [hunk 1]
```diff
@@ -23,10 +31,53 @@ def talk(request, talk_slug):
...
+    talk_update_form = TalkUpdateForm(instance=talk)
+
+    if request.method == "POST":
+        talk_update_form = TalkUpdateForm(request.POST, instance=talk)
+
+        if talk_update_form.is_valid():
+            talk_update_form.save(request.user)
+            messages.success(request, "Talk details updated")
+            return redirect(talk.get_absolute_url())
+        else:
```
Comment (artcz) [inline conference/talks.py:58]: else is not needed because return inside the if block.

File: conference/talks.py [hunk 2]
```diff
@@ -23,10 +31,53 @@ def talk(request, talk_slug):
...
+def can_user_update_talk(user, talk):
+    """
+    Check if the user can update the given talk.
+    """
+    conf = Conference.objects.current()
+
+    return (
+        talk.created_by == user
+        and talk.status == TALK_STATUS.accepted
+        and not conf.has_finished
```
Comment (artcz) [inline conference/talks.py:77]: that conf.has_finished part is tricky, depends on how we'd want to handle the slide upload if it happens after the conference, but let's keep it as is for now.

File: conference/talks.py [hunk 3]
```diff
@@ -23,10 +31,53 @@ def talk(request, talk_slug):
...
+
+def can_user_update_talk(user, talk):
+    """
+    Check if the user can update the given talk.
+    """
+    conf = Conference.objects.current()
+
+    return (
+        talk.created_by == user
+        and talk.status == TALK_STATUS.accepted
```
Comment (artcz) [inline conference/talks.py]: Q: are we still able to edit proposals during CFP?

File: conference/talks.py [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
+from django.contrib.auth.decorators import login_required
 from django.core.urlresolvers import reverse
-from django.shortcuts import get_object_or_404
+from django.http import HttpResponseForbidden
+from django.shortcuts import get_object_or_404, redirect
 from django.template.response import TemplateResponse
 
-from conference.models import Talk
+from conference.models import Conference, Talk, TALK_STATUS
+from conference.new_forms import TalkUpdateForm
```
Comment (artcz) [inline conference/talks.py]: I'm a bit sceptical about the name. Maybe `conference.talk_forms`? (or `forms/talks.py`) WDYT?

**Review by artcz — APPROVED** [review]
LGTM, with one comment about forms module name.

File: conference/talks.py (hunk 2 above)
Comment (umgelurgel) [reply conference/talks.py:77]: Sounds good; the slide upload sounds like a candidate for a separate form + view.

File: conference/talks.py (hunk 4 above)
Comment (umgelurgel) [reply conference/talks.py]: I like `forms/talks.py` - I've called it new_forms as I didn't want to add to the existing `forms.py`, most of which is due to be retired after ep2019.

**Pull request merged**